### PR TITLE
Adding another handful transform's extensions

### PR DESCRIPTION
**Pull request by sfilipi**
Part of #1318 

Adds more transform extensions in the catalog.

### CODE REVIEW

File: src/Microsoft.ML.Data/Transforms/ConversionsCatalog.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public static class HashingEstimatorCatalog
         /// <param name="outputColumn">Name of the column to be transformed. If this is null '<paramref name="inputColumn"/>' will be used.</param>
         /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 31, inclusive.</param>
         /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
-        public static HashingEstimator Hash(this TransformsCatalog.Conversions catalog, string inputColumn, string outputColumn = null,
+        public static HashingEstimator Hash(this TransformsCatalog.ConversionTransforms catalog, string inputColumn, string outputColumn = null,
```
Comment (singlis) [inline src/Microsoft.ML.Data/Transforms/ConversionsCatalog.cs]: nit - Why rename this? TransformsCatalog.ConversionTransforms sounds a bit redundant as this sounds like there is a TransformsCatalog and that Conversions is related to transforms that deal with converting.

File: src/Microsoft.ML.PCA/PCACatalog.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Transforms.Projections;
+
+namespace Microsoft.ML
+{
+    public static class PcaCatalog
+    {
+
+        /// <summary>Initializes a new instance of <see cref="PrincipalComponentAnalysisEstimator"/>.</summary>
```
Comment (singlis) [inline src/Microsoft.ML.PCA/PCACatalog.cs]: ></summary> [](start = 101, length = 10)

Not needed.

File: src/Microsoft.ML.Transforms/ExtensionsCatalog.cs [hunk 3]
```diff
@@ -32,6 +32,31 @@ public static class MissingValueIndicatorCatalog
...
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="inputColumn">The name of the input column.</param>
+        /// <param name="outputColumn">The optional name of the output column,
+        /// If not provided, the <paramref name="inputColumn"/> will be replaced with the results of the transforms.</param>
+        /// <param name="replacementKind">The type of replacement to use as specified in <see cref="NAReplaceTransform.ColumnInfo.ReplacementMode"/></param>
+        public static MissingValueReplacingEstimator ReplaceMissingValues(this TransformsCatalog catalog,
+            string inputColumn,
+            string outputColumn = null,
+            NAReplaceTransform.ColumnInfo.ReplacementMode replacementKind = MissingValueReplacingEstimator.Defaults.ReplacementMode)
+        => new MissingValueReplacingEstimator(CatalogUtils.GetEnvironment(catalog), inputColumn, outputColumn, replacementKind);
```
Comment (singlis) [inline src/Microsoft.ML.Transforms/ExtensionsCatalog.cs]: tabbing here.

File: src/Microsoft.ML.Transforms/ExtensionsCatalog.cs [hunk 4]
```diff
@@ -32,6 +32,31 @@ public static class MissingValueIndicatorCatalog
...
 
+    public static class MissingValueReplacerCatalog
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="MissingValueReplacingEstimator"/>
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="inputColumn">The name of the input column.</param>
+        /// <param name="outputColumn">The optional name of the output column,
+        /// If not provided, the <paramref name="inputColumn"/> will be replaced with the results of the transforms.</param>
```
Comment (singlis) [inline src/Microsoft.ML.Transforms/ExtensionsCatalog.cs]: >inputColumn [](start = 49, length = 11)

Is this correct? The input column will be replaced with the results? Or should this be outputColumn?

File: src/Microsoft.ML.Transforms/ProjectionCatalog.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    public static class ProjectionCatalog
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="RandomFourierFeaturizingEstimator"/>.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="inputColumn">Name of the column to be transformed.</param>
+        /// <param name="outputColumn">Name of the output column. If this is null '<paramref name="inputColumn"/>' will be used.</param>
+        /// <param name="newDim">The number of random Fourier features to create.</param>
+        /// <param name="useSin">Create two features for every random Fourier frequency? (one for cos and one for sin).</param>
```
Comment (singlis) [inline src/Microsoft.ML.Transforms/ProjectionCatalog.cs]: >frequency? [](start = 78, length = 10)

is the ? supposed to be there?

File: src/Microsoft.ML.Data/Transforms/ConversionsCatalog.cs (hunk 1 above)
Comment (sfilipi) [reply src/Microsoft.ML.Data/Transforms/ConversionsCatalog.cs]: If you look at the TransformsCatalog, i standardize the naming there. This is an effect of that.  

---
In reply to: [230153450](https://github.com/dotnet/machinelearning/pull/1494#discussion_r230153450) [](ancestors = 230153450)

File: src/Microsoft.ML.Transforms/ExtensionsCatalog.cs (hunk 4 above)
Comment (sfilipi) [reply src/Microsoft.ML.Transforms/ExtensionsCatalog.cs]: it is correct. The inputColumn will be overriten:


  public MissingValueReplacingEstimator(IHostEnvironment env, string inputColumn, string outputColumn = null, NAReplaceTransform.ColumnInfo.ReplacementMode replacementKind = Defaults.ReplacementMode)
            : this(env, new NAReplaceTransform.ColumnInfo(outputColumn ?? inputColumn, inputColumn, replacementKind))

---
In reply to: [230155013](https://github.com/dotnet/machinelearning/pull/1494#discussion_r230155013) [](ancestors = 230155013)

File: src/Microsoft.ML.PCA/PCACatalog.cs (hunk 2 above)
Comment (sfilipi) [reply src/Microsoft.ML.PCA/PCACatalog.cs]: why?  All the examples close the tag:

[https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/xml-documentation-comments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/xml-documentation-comments)

---
In reply to: [230154186](https://github.com/dotnet/machinelearning/pull/1494#discussion_r230154186) [](ancestors = 230154186)

**Pull request merged**